core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -1108,17 +700,18 @@ public int next() {
 
         DescendingIter() {
             from = null;
-            stack = new IntStack(entries.length / FIELDS);
+            stack = new IntStack(entrySet.getNumOfEntries());
             int sortedCnt = sortedCount.get();
-            anchor = sortedCnt == 0 ? HEAD_NODE : (sortedCnt - 1) * (FIELDS) + 1; // this is the last sorted entry
+            anchor = // this is the last sorted entry
+                (sortedCnt == 0 ? entrySet.getNextEntryIndex(entrySet.getHeadNextIndex()) : sortedCnt);
```

Comment:
I believe that the last sorted entry is in index `sortedCnt-1`, not in index `sortedCnt`.

Reply:
first entry has entry index 1 not 0